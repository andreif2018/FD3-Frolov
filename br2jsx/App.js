"use strict";

import React from 'react';
import ReactDOM from 'react-dom';
import MainBlock from "../br2jsx/components/MainBlock";

ReactDOM.render(
    <MainBlock/>, document.getElementById('container')
);