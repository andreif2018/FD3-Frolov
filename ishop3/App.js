"use strict";

import React from 'react';
import ReactDOM from 'react-dom';
import ShopBlock from './components/ShopBlock';
var shopHeaders = require('./headers.json');
var shopArr = require('./productList.json');
var shopText = "Список товаров в интернет магазине";

ReactDOM.render(
    React.createElement(ShopBlock, { title: shopText, headers: shopHeaders, items: shopArr}),
    document.getElementById("container")
);
