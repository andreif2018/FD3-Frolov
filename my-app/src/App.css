@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}
a {
  color: white;
  text-decoration: none;
}
.active {
  padding-bottom: 5px;
  border-bottom: 2px solid cyan;
  text-decoration: darkcyan;
  text-shadow: mediumseagreen 1px 0 7px;
}
.App-header {
  background-color: #282c34;
  min-height: calc(10px + 2vmin);
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-around;
  font-size: calc(10px + 2vmin);
  padding: 30px;
}
.App-footer {
  background-color: #282c34;
  min-height: calc(10px + 2vmin);
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-around;
  font-size: calc(10px + 1vmin);
  color: white;
  padding: 30px;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
