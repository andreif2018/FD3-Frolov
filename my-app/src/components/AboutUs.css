div.AboutUs {
    flex: 1;
    background-color: dimgrey;
    text-align: center;
    color: white;
    margin-top: 0;
    font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New',
    monospace;
}
p {
    margin-top: 0;
    padding: 20px;
}

@media only screen and (min-width: 1400px) {
    div.AboutUs p{
        font-size: 1.2em;
    }
}
@media only screen and (max-width: 1400px) {
    div.AboutUs p {
        font-size: 1.1em;
    }
}
@media only screen and (max-width: 960px) {
    div.AboutUs p {
        font-size: 1em;
    }
}
@media only screen and (max-width: 760px) {
    div.AboutUs p {
        font-size: 0.9em;
    }
}
@media only screen and (max-width: 500px) {
    div.AboutUs p {
        font-size: 0.8em;
    }
}