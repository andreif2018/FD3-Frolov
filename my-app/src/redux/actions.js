export const ADD_SONG = "ADD_SONG";
export const DELETE_SONG = "DELETE_SONG";
export const RESET = "RESET";
export const SAVE_PLAYLIST = "SAVE_PLAYLIST";
export const DELETE_PLAYLIST = "DELETE_PLAYLIST";

export const addSong = content => ({
    type: ADD_SONG,
    payload: content
});

export const deleteSong = content => ({
    type: DELETE_SONG,
    payload: content
});

export const reset = () => ({
    type: RESET
});

export const savePlaylist = (content) => ({
    type: SAVE_PLAYLIST,
    payload: content
});

export const deletePlaylist = content => ({
    type: DELETE_PLAYLIST,
    payload: content
});

