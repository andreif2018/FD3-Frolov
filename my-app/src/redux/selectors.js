import { VISIBILITY_FILTERS } from "../constants";

export const getSongsState = store => store.songs;

export const getSongList = store =>
    getSongsState(store) ? getSongsState(store) : [];

/**
 * example of a slightly more complex selector
 * select from store combining information from multiple reducers
 */
export const getSongs = store =>
    getSongList(store);

export const getSongsByVisibilityFilter = (store, visibilityFilter) => {
    const allSongs = getSongs(store);
    switch (visibilityFilter) {
        case VISIBILITY_FILTERS.BLUES:
            return allSongs.filter(song => song.genre === "Blues");
        case VISIBILITY_FILTERS.ROCK:
            return allSongs.filter(song => song.genre === "Rock");
        case VISIBILITY_FILTERS.JAZZ:
            return allSongs.filter(song => song.genre === "Jazz");
        case VISIBILITY_FILTERS.ALL:
        default:
            return allSongs;
    }
};
