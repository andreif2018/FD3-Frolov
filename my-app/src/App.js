import './App.css';
import logo from "./logo.png";
import AboutUs from "./components/AboutUs";
import {
    Switch,
    Route,
    NavLink,
    Redirect
} from "react-router-dom";
import AllMusic from "./components/AllMusic";
import Playlist from "./components/Playlist";
import { BrowserRouter } from "react-router-dom";
import MyLib from "./components/MyLib";
import Badge from 'react-simple-badges'
import * as React from "react";
import {myEvents} from "./components/events";
import SubPage from "./components/SubPage";
import items from "./components/songList.json";

class App extends React.PureComponent {

    constructor(props) {
        super(props);
        this.updateWindowDimensions = this.updateWindowDimensions.bind(this);
    }

    state = {
        badge: 'none',
        width: 0,
        menuState: 'none',
    }

    updateWindowDimensions() {
        this.setState({ width: window.innerWidth });
    }

    addBadge = () => {
        this.setState({badge: "inline"});
    };

    removeBadge = () => {
        this.setState({badge: "none"});
    };

    openCloseBurger = () => {
        if (this.state.menuState === 'none') this.setState({menuState: 'block'});
        else if (this.state.menuState === 'block') this.setState({menuState: 'none'});
    }

    componentDidMount = () => {
        myEvents.addListener('UpdateBadge', this.addBadge);
        myEvents.addListener('RemoveBadge', this.removeBadge);
        this.updateWindowDimensions();
        window.addEventListener('resize', this.updateWindowDimensions);
    };

    componentWillUnmount = () => {
        myEvents.removeListener('UpdateBadge', this.addBadge);
        myEvents.addListener('RemoveBadge', this.removeBadge);
        window.removeEventListener('resize', this.updateWindowDimensions);
    };

    render() {
        let mobileBurgerClassName = (window.innerWidth > 500) ? "BurgerWeb" : "BurgerMobile";
        let appHeaderClassName = (window.innerWidth > 500) ? "App-header" : "AppHeaderMobile";
        return (
            <BrowserRouter>
                <div className="App-logo"><img src={logo} alt={"logo of the app"}/></div>
                <header className={appHeaderClassName}>
                    <NavLink to="/allMusic">All Music</NavLink>
                    <NavLink to="/myLib">My Library</NavLink>
                    <NavLink to="/playList">PlayList
                        <Badge name="badge" label={"new"} style={{borderRadius: '50%', display: this.state.badge}}/>
                    </NavLink>
                    <NavLink to="/about">About Us</NavLink>
                </header>
                <div className={mobileBurgerClassName}>
                    <img className="Burger" onClick={this.openCloseBurger}
                        src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAACAAAAAgCAYAAABzenr0AAAARUlEQVRYR+3U0QkAIAgFQN1/6IL6aQIVOhd4cujLaJ5szg8LEHgFVvFBnmwLEBglUPwEN25UEekBAgT0gBsgQEAP/CmwAdmEEiEL49BKAAAAAElFTkSuQmCC"
                        alt="Открытие меню" title="Открытие меню"/><br/><br/>
                        <div style={{display: this.state.menuState}}>
                            <NavLink to="/allMusic">All Music</NavLink><br/><br/>
                            <NavLink to="/myLib">My Library</NavLink><br/><br/>
                            <NavLink to="/playList">PlayList
                                <Badge name="badge" label={"new"} style={{borderRadius: '50%', display: this.state.badge}}/>
                            </NavLink><br/><br/>
                            <NavLink to="/about">About Us</NavLink>
                        </div>

                </div>
                <Switch>
                    <Route exact path="/">
                        <Redirect to="/about"/>
                    </Route>
                    <Route path="/allMusic">
                        <AllMusic/>
                    </Route>
                    <Route path="/myLib">
                        <MyLib/>
                    </Route>
                    <Route path="/playList">
                        <Playlist/>
                    </Route>
                    <Route path="/about">
                        <AboutUs/>
                    </Route>
                    <Route path="/allMusic1">
                        <SubPage key={1} items={items.slice(0, 20)}/>
                    </Route>
                    <Route path="/allMusic2">
                        <SubPage key={2} items={items.slice(20, 40)}/>
                    </Route>
                    <Route path="/allMusic3">
                        <SubPage key={3} items={items.slice(40)}/>
                    </Route>
                </Switch>
                <div className="App-footer">This is MyApp</div>
            </BrowserRouter>
        );
    }
}
export default App;
