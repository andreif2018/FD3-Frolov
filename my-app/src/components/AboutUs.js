import React from 'react';
import './AboutUs.css';

class AboutUs extends React.PureComponent {

    render() {
        return (
            <div className="AboutUs">
                <p>This app allows you to create your own playlists.</p>
                <p>Start to add items to playlist on "All music" page.</p>
                <p>Then navigate to "Playlist" page to save created playlist</p>
                <p>Also, You will be able to filter your songs by genre.</p>
            </div>
        );
    }
}

export default AboutUs;