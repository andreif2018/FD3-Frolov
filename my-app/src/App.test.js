import renderer from 'react-test-renderer';
import AboutUs from "./components/AboutUs";
import {ADD_SONG, DELETE_PLAYLIST, DELETE_SONG, SAVE_PLAYLIST} from "./redux/actions";
import * as actions from "./redux/actions";
import * as React from "react";
import App from "./App";

test('About Us page is default state for opening app', () => {
  const component = renderer.create(<AboutUs/>);
  let componentTree=component.toJSON();
  expect(componentTree).toMatchSnapshot('AboutUsDefault.test.js.snap');
});

/* redux testing start*/
describe('adding Song to playlist', () => {
  it('should create an action to add a song', () => {
    const payload = {"code":1, "song":"Blueberry Hill", "artist":"Louis Armstrong", "album": "Hello, Dolly!", "year":1964, "genre":"Jazz"};
    const expectedAction = {
      type: ADD_SONG,
      payload
    }
    expect(actions.addSong(payload)).toEqual(expectedAction)
  })
});

describe('deletion Song from playlist', () => {
  it('should create an action to delete a song', () => {
    const payload = {"code":2, "song":"One After 909", "artist":"The Beatles", "album": "Let It be", "year":1970, "genre":"Rock"};
    const expectedAction = {
      type: DELETE_SONG,
      payload
    }
    expect(actions.deleteSong(payload)).toEqual(expectedAction)
  })
});

describe('store certain PlayList into MyLib', () => {
  it('should create an action to save playlist', () => {
    const payload = [
      "playlistName",
      "1",
      [
        {
          "code": 1,
          "song": "Blueberry Hill",
          "artist": "Louis Armstrong",
          "album": "Hello, Dolly!",
          "year": 1964,
          "genre": "Jazz",
          "isFiltered": false
        }
      ]
    ];
    const expectedAction = {
      type: SAVE_PLAYLIST,
      payload
    }
    expect(actions.savePlaylist(payload)).toEqual(expectedAction)
  })
});

describe('delete certain PlayList from MyLib', () => {
  it('should create an action to delete playlist', () => {
    const payload = [
      "playlistName",
      "1",
      [
        {
          "code": 1,
          "song": "Blueberry Hill",
          "artist": "Louis Armstrong",
          "album": "Hello, Dolly!",
          "year": 1964,
          "genre": "Jazz",
          "isFiltered": false
        }
      ]
    ];
    const expectedAction = {
      type: DELETE_PLAYLIST,
      payload
    }
    expect(actions.deletePlaylist(payload)).toEqual(expectedAction)
  })
});
/* redux testing end*/