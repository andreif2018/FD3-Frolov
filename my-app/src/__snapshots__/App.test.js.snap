// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Us default state: AboutUsDefault.test.js.snap 1`] = `
<div
  className="AboutUs"
>
  <p>
    This app allows you to create your own playlists.
  </p>
  <p>
    Start to add items to playlist on "All music" page.
  </p>
  <p>
    Then navigate to "Playlist" page to save created playlist
  </p>
  <p>
    Also, You will be able to filter your songs by genre.
  </p>
</div>
`;

exports[`About Us page is default state for opening app: AboutUsDefault.test.js.snap 1`] = `
<div
  className="AboutUs"
>
  <p>
    This app allows you to create your own playlists.
  </p>
  <p>
    Start to add items to playlist on "All music" page.
  </p>
  <p>
    Then navigate to "Playlist" page to save created playlist
  </p>
  <p>
    Also, You will be able to filter your songs by genre.
  </p>
</div>
`;
